interference between Timer and R_useDynamicSymbols

I have a very simple 📦 here https://github.com/romainfrancois/testtimer to demonstrate this issue. 

The relevant thing is that it includes `Timer.h` in its `_types.h` file : 

```
#include <Rcpp.h>
#include <Rcpp/Benchmark/Timer.h>
```

and so when I install it I get: 

```r
> install_github("romainfrancois/testtimer")
Downloading GitHub repo romainfrancois/testtimer@master
from URL https://api.github.com/repos/romainfrancois/testtimer/zipball/master
Installing testtimer
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/r_/1b2gjtsd7j92jbbpz4t7ps340000gn/T/Rtmpz9TfM1/devtools40674e600f2/romainfrancois-testtimer-cd72001'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library' --with-keep.source --install-tests 

* installing *source* package ‘testtimer’ ...
** libs
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/Rcpp/include" -I/usr/local/include   -fPIC  -Wno-unused-result -Wno-c++11-inline-namespace -O3 -c RcppExports.cpp -o RcppExports.o
RcppExports.cpp:26:5: error: no matching function for call to 'R_useDynamicSymbols'
    R_useDynamicSymbols(dll, FALSE);
    ^~~~~~~~~~~~~~~~~~~
/Library/Frameworks/R.framework/Resources/include/R_ext/Rdynload.h:84:10: note: candidate function not viable: no known conversion from 'int' to 'Rboolean' for 2nd argument
Rboolean R_useDynamicSymbols(DllInfo *info, Rboolean value);
         ^
1 error generated.
make: *** [RcppExports.o] Error 1
ERROR: compilation failed for package ‘testtimer’
* removing ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/testtimer’
Installation failed: Command failed (1)
```

macOS specific I fear:

```bash
devtools::install_github("romainfrancois/testtimer")
Downloading GitHub repo romainfrancois/testtimer@master
from URL https://api.github.com/repos/romainfrancois/testtimer/zipball/master
Installing testtimer
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/b0/vt_1hj2d6yd8myx9lwh81pww0000gn/T/RtmpjLr5Qf/devtoolsf2ae3b74c003/romainfrancois-testtimer-cd72001'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library' --install-tests 

* installing *source* package ‘testtimer’ ...
** libs
/usr/local/clang4/bin/clang++ -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c RcppExports.cpp -o RcppExports.o
RcppExports.cpp:26:5: error: no matching function for call to 'R_useDynamicSymbols'
    R_useDynamicSymbols(dll, FALSE);
    ^~~~~~~~~~~~~~~~~~~
/Library/Frameworks/R.framework/Resources/include/R_ext/Rdynload.h:84:10: note: candidate function not viable: no known conversion from 'int' to 'Rboolean' for 2nd argument
Rboolean R_useDynamicSymbols(DllInfo *info, Rboolean value);
         ^
1 error generated.
make: *** [RcppExports.o] Error 1
ERROR: compilation failed for package ‘testtimer’
* removing ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/testtimer’
Installation failed: Command failed (1)
```

For good measure I also tossed it at WinBuilder and it fails there too:

```
* installing *source* package 'testtimer' ...
** libs

*** arch - i386
d:/Compiler/gcc-4.9.3/mingw_32/bin/g++  -I"D:/RCompile/recent/R-3.5.0/include" -DNDEBUG  -I"d:/RCompile/CRANpkg/lib/3.5/Rcpp/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c RcppExports.cpp -o RcppExports.o
RcppExports.cpp: In function 'void R_init_testtimer(DllInfo*)':
RcppExports.cpp:26:35: error: invalid conversion from 'int' to 'Rboolean' [-fpermissive]
     R_useDynamicSymbols(dll, FALSE);
                                   ^
In file included from D:/RCompile/recent/R-3.5.0/include/Rinternals.h:51:0,
                 from d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp/r/headers.h:53,
                 from d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/RcppCommon.h:38,
                 from d:/RCompile/CRANpkg/lib/3.5/Rcpp/include/Rcpp.h:27,
                 from ../inst/include/testtimer_types.h:1,
                 from RcppExports.cpp:4:
D:/RCompile/recent/R-3.5.0/include/R_ext/Rdynload.h:84:10: note: initializing argument 2 of 'Rboolean R_useDynamicSymbols(DllInfo*, Rboolean)'
 Rboolean R_useDynamicSymbols(DllInfo *info, Rboolean value);
          ^
make: *** [RcppExports.o] Error 1
ERROR: compilation failed for package 'testtimer'
* removing 'd:/RCompile/CRANguest/R-release/lib/testtimer'
In R CMD INSTALL
```

--- a/inst/include/Rcpp/Benchmark/Timer.h
+++ b/inst/include/Rcpp/Benchmark/Timer.h
@@ -120,17 +120,17 @@ namespace Rcpp{
             out.attr("names") = names;
             return out;
         }
-        
+
         static std::vector<Timer> get_timers(int n){
             return std::vector<Timer>( n, Timer() ) ;
         }
-        
+
         inline nanotime_t now() const {
             return get_nanotime() ;
         }
-        
+
         inline nanotime_t origin() const {
-            return start_time ;    
+            return start_time ;
         }
 
     private:
@@ -143,5 +143,8 @@ namespace Rcpp{
 
 }
 
+#ifdef FALSE
+  #undef FALSE
 #endif
 
+#endif

